### PR TITLE
[W-11149125 & W-11125193] - Edit Payment Details for RD2

### PR DESCRIPTION
**Pull request by JoseRossiSF**
# Technical Document
[Technical Design Review: W-11149125 & W-11125193 - NPSP](https://salesforce.quip.com/Yd8BA0z4EAFY)

# Changes
Passing the allowACHPaymentMethod attribute from rd2RecurringDonation table component to updatePaymentMethodModal child component alongside the currentRecord attribute.
Checking in child component whether ACH Payment is allowed and what is the payment method for the current record, and will load the payment options dynamically into a lightning-radio-group element.

# Work Items
[W-11149125 ](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000x9auaYAA/view)& [W-11125193](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000wnLdDYAU/view)

### CODE REVIEW

File: force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js [hunk 1]
```diff
@@ -8,15 +8,46 @@ const TAB_KEY_CODE = 9;
...
-      this.template.addEventListener("keydown", (e) => this.handleKeyUp(e));
+        this.template.addEventListener("keydown", (e) => this.handleKeyUp(e));
+        if(this.currentRecord){
+            this.determineACHpaymentMethodAndAddAsOption();
+            this.paymentMethodValue = this.currentRecord.paymentMethod;
+        }
+    }
+
+    determineACHpaymentMethodAndAddAsOption(){
+        if(this.isBankPaymentAllowed){
```
Comment (lparrott) [inline force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js:40]: Couldn't these all be combined into one `if` condition? It looks like the body of each `if` block is the same logic.

File: force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js [hunk 2]
```diff
@@ -8,15 +8,46 @@ const TAB_KEY_CODE = 9;
...
 
 export default class UpdatePaymentMethodModal extends LightningElement {
+    
+    @api isBankPaymentAllowed;
+
     @api openUpdatePaymentMethod;
+    
     @api currentRecord;
 
+    @track
```
Comment (lparrott) [inline force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js:18]: I don't think `@track` is needed anymore, but I could be wrong!

**Review by lparrott — CHANGES_REQUESTED** [review]

**Review by gerardorodriguezsf — CHANGES_REQUESTED** [review]
- Prevent duplicated actions due to the rendered actions.
- Please do not user hardcoded 'ACH', RD2_Constants.PAYMENT_PICKLIST_VALUE_ACH

File: force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js (hunk 1 above)
Comment (JoseRossiSF) [reply force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js]: done!

**Review by lparrott — APPROVED** [review]

File: force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js [hunk 3]
```diff
@@ -1,22 +1,54 @@
...
 
+    paymentMethodValue = "";
+
     labels = {
       updatePaymentMethod
     }   
 
+    @track
+    paymentMethodOptions = [
+        { label: 'Credit Card', value: 'Credit Card' }
```
Comment (maurimart) [inline force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js:27]: value and label are repeated, shouldn't we use a custom label for these?

File: force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js [hunk 4]
```diff
@@ -1,22 +1,54 @@
...
+                this.determineACHpaymentMethodAndAddAsOption();
+                this.paymentMethodValue = this.currentRecord.paymentMethod;
+            }
+        }
+    }
+
+    determineACHpaymentMethodAndAddAsOption(){
+        if(this.isBankPaymentAllowed || this.currentRecord.paymentMethod === "ACH"){
+            if( !this.paymentMethodOptions.some( element => element.value === 'ACH') ){
+                this.paymentMethodOptions.push( { label: 'Bank Account', value: ACH_PAYMENT_METHOD } );
```
Comment (maurimart) [inline force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js:47]: shouldn't we use a custom label for `Bank Account`?

File: force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js [hunk 5]
```diff
@@ -1,22 +1,54 @@
...
+            }
+        } else if (this.paymentMethodOptions.some( element => element.value === 'ACH')){
+            this.paymentMethodOptions.pop( { label: 'Bank Account', value: ACH_PAYMENT_METHOD } );
```
Comment (maurimart) [inline force-app/main/default/lwc/updatePaymentMethodModal/updatePaymentMethodModal.js:50]: same for this label, repeated and hardcoded text.

**Review by maurimart — CHANGES_REQUESTED** [review]

**Review by maurimart — APPROVED** [review]
LGTM

**Review by gerardorodriguezsf — APPROVED** [review]
LGTM

**Review by gerardorodriguezsf — APPROVED** [review]

**Pull request merged**